staging/src/k8s.io/api/networking/v1/types.go

```diff
@@ -302,16 +302,16 @@ type IngressList struct {
 
 // IngressSpec describes the Ingress the user wishes to exist.
 type IngressSpec struct {
-	// IngressClassName is the name of the IngressClass cluster resource. The
-	// associated IngressClass defines which controller will implement the
-	// resource. This replaces the deprecated `kubernetes.io/ingress.class`
-	// annotation. For backwards compatibility, when that annotation is set, it
-	// must be given precedence over this field. The controller may emit a
-	// warning if the field and annotation have different values.
-	// Implementations of this API should ignore Ingresses without a class
-	// specified. An IngressClass resource may be marked as default, which can
-	// be used to set a default value for this field. For more information,
-	// refer to the IngressClass documentation.
+	// IngressClassName is the name of an IngressClass cluster resource. Ingress
+	// controller implementations use this field to know whether they should be
+	// serving this Ingress resource, by a transitive connection
+	// (controller -> IngressClass -> Ingress resource). Although the
+	// `kubernetes.io/ingress.class` annotation (simple constant name) was never
+	// formally defined, it was widely supported by Ingress controllers to create
+	// a direct bound between Ingress controller and Ingress resources. Newly
+	// created Ingress resources should prefer using the field. However, even
+	// though the annotation is officially deprecated, for backwards compatibility
+	// reasons the annotation should still be honored if present.
```

Comment:
Maybe it would be good to tell how the linking should be done.
I guess ingress controller reads `IngressClass.spec.controller`, but it could be also `IngressClass.metadata.name` to identify what kind of `ingress.spec.ingressClassName` to be responsible for.

Reply:
I like this.

Reply:
> Maybe it would be good to tell how the linking should be done.
> I guess ingress controller reads `IngressClass.spec.controller`, but it could be also `IngressClass.metadata.name` to identify what kind of `ingress.spec.ingressClassName` to be responsible for.

I guess this is more something for the IngressClass API Docs or even the k8s.io docs ?!

Reply:
I am OK with the idea of having this information part of the Kubernetes documentation.

**For later:** it would be great if, after having updated the Kubernetes documentation, we could add a link to the end of this API documentation so that people can jump straight to the right documentation page and read more about how they should expect the ingress controllers to behave.